### PR TITLE
Change displayname of user as admin in rooms

### LINKED ISSUE #6572: Change displayname of users as admin does not work
### Description

I try to change the displayname of other users as admin. The change in profile ist ok, but I get errors for displayname in rooms.

```
2019-12-19 14:45:57,831 - synapse.access.http.8008 - 233 - INFO - PUT-906- - - 8008 - Received request: PUT /_matrix/client/r0/profile/@user:server.cloudapp.azure.com/displayname
2019-12-19 14:45:57,932 - synapse.handlers.message - 695 - WARNING - PUT-906- Denying new event <FrozenEventV3 event_id='$e_mVDdJ-rQVW25vaOgXIU3zNppPZfOCJUo5YhMQ0rTM', type='m.room.member', state_key='@user:server.cloudapp.azure.com'> because 403: Cannot force another user to join.
2019-12-19 14:45:57,932 - synapse.handlers.profile - 281 - WARNING - PUT-906- Failed to update join event for room !GstgHdhajwAvSueHan:server.cloudapp.azure.com - 403: Cannot force another user to join.
2019-12-19 14:45:57,966 - synapse.handlers.message - 695 - WARNING - PUT-906- Denying new event <FrozenEvent event_id='$157676675735pBzAC:server.cloudapp.azure.com', type='m.room.member', state_key='@user:server.cloudapp.azure.com'> because 403: Cannot force another user to join.
2019-12-19 14:45:57,967 - synapse.handlers.profile - 281 - WARNING - PUT-906- Failed to update join event for room !AvSWOczTtSmEVpjVlF:server.cloudapp.azure.com - 403: Cannot force another user to join.
2019-12-19 14:45:58,724 - synapse.access.http.8008 - 302 - INFO - PUT-906- - - 8008 - {@admin_user:server.cloudapp.azure.com} Processed request: 0.892sec/0.000sec (0.051sec, 0.011sec) (0.085sec/0.662sec/109) 3B 200 "PUT /_matrix/client/r0/profile/@user:server.cloudapp.azure.com/displayname HTTP/1.1" "curl/7.58.0" [0 dbevts]
```

### Steps to reproduce

```
curl -X PUT --header 'Authorization: Bearer access_token' --header 'Content-Type: application/json' --header 'Accept: application/json' -d '{ "displayname": "new name" }' 'http://localhost:8008/_matrix/client/r0/profile/@user:server.cloudapp.azure.com/displayname' -i

HTTP/1.1 200 OK
Server: Synapse/1.6.1
Date: Thu, 19 Dec 2019 14:45:57 GMT
Content-Type: application/json
Content-Length: 3
Cache-Control: no-cache, no-store, must-revalidate
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept, Authorization

{}
```

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**: 

If not matrix.org:

<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- **Version**: v1.6.1

- **Install method**: Ubuntu/Debian package manager
<!-- examples: package manager/git clone/pip  -->

- **Platform**: Ubuntu, Azure VM
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->


How can I support? Do you need more informations?
Callled function: https://github.com/matrix-org/synapse/blob/master/synapse/handlers/profile.py#L257
Error: https://github.com/matrix-org/synapse/blob/master/synapse/event_auth.py#L294

### PR DESCRIPTION
**Pull request by dklimpel**
Try to fix #6572

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file]
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: synapse/handlers/profile.py [hunk 1]
```diff
@@ -165,6 +165,12 @@ def set_displayname(self, target_user, requester, new_displayname, by_admin=Fals
         if new_displayname == "":
             new_displayname = None
 
+        # If the admin changes the display name of a user, the user cannot send
+        # the join event to update the displayname in the rooms.
+        # This must be done by the user himself.
+        if by_admin:
+            requester = create_requester(target_user.to_string())
```
Comment (richvdh) [inline synapse/handlers/profile.py]: `create_requester` will accept a `UserID` object

```suggestion
            requester = create_requester(target_user)
```

File: synapse/handlers/profile.py [hunk 2]
```diff
@@ -165,6 +165,12 @@ def set_displayname(self, target_user, requester, new_displayname, by_admin=Fals
         if new_displayname == "":
             new_displayname = None
 
+        # If the admin changes the display name of a user, the user cannot send
```
Comment (richvdh) [inline synapse/handlers/profile.py]:
```suggestion
        # If the admin changes the display name of a user, the requesting user cannot send
```

File: synapse/handlers/profile.py [hunk 3]
```diff
@@ -165,6 +165,12 @@ def set_displayname(self, target_user, requester, new_displayname, by_admin=Fals
...
+        # the join event to update the displayname in the rooms.
+        # This must be done by the user himself.
```
Comment (richvdh) [inline synapse/handlers/profile.py]:
```suggestion
        # This must be done by the target user himself.
```

File: changelog.d/6572.bugfix [hunk 4]
```diff
@@ -0,0 +1 @@
+Change displayname of users as admin in rooms.
```
Comment (richvdh) [inline changelog.d/6572.bugfix]: could you add some more words here? remember that the changelog is supposed to explain to Synapse admins what has changed: this really doesn't. Also feel free to add an attribution to yourself.

Something like this?

> When a user's profile is updated via the admin API, also generate a displayname/avatar update for that user in each room. Contributed by @dklimpel.

Comment (dklimpel) [reply changelog.d/6572.bugfix]: Thanks for your suggestion and review.

**Pull request merged**